### PR TITLE
Update: GitHub Actions workflow to remove the bot comment when Trac Ticket is added to the PR.

### PR DESCRIPTION
**Pull request by Debarghya-Banerjee**
**Trac Ticket - [Core-61567](https://core.trac.wordpress.org/ticket/61567)**

## Problem Statement:
Bot comments reminding contributors to include Trac ticket links in pull requests accumulate over time, cluttering discussion threads even after the links are added. This clutter hinders efficient review processes and detracts from meaningful collaboration in WordPress core development.

## Solution:

- Implement automated checks to:
    - Verify each pull request includes a link to a corresponding Trac ticket.
    - Create a notice if a Trac ticket link is missing.
    - Automatically remove the notice upon adding a valid Trac ticket link to streamline pull request review processes and reduce unnecessary bot comments.

## Impact:

- Automating the deletion of bot comments once the Trac ticket link is updated in GitHub pull requests for WordPress core development provides several key benefits:

    - `Enhanced Review Efficiency`: Streamlines PR discussions by reducing clutter, allowing faster decision-making and clearer communication.
    - `Improved Collaboration`: Encourages focused participation and feedback, promoting a more engaged development process.
    - `Simplified Workflow`: Enhances user experience by minimizing distractions and optimizing contributor workflows.
    - `Overall Efficiency`: Optimizes reviewer resources, maximizing productivity and the impact of community contributions.

### CODE REVIEW

File: .github/workflows/pull-request-comments.yml [hunk 1]
```diff
@@ -164,16 +164,16 @@ jobs:
 
             github.rest.issues.createComment( commentInfo );
 
-  # Leaves a comment on a pull request when no Trac ticket is included in the pull request description.
+  # Leaves a comment on a pull request when no Trac ticket is included in the pull request description and delete the comment if the Trac ticket has been updated.
   trac-ticket-check:
-    name: Comment on a pull request when no Trac ticket is included
+    name: Comment on a pull request when no Trac ticket is included and delete the comment when Trac Ticket has been updated
```
Comment (desrosj) [inline .github/workflows/pull-request-comments.yml]: Let's adjust this a bit. It's possible that more tasks are added to this in the future. So let's be more general for the `name` and move the details into the description.

Comment (desrosj) [inline .github/workflows/pull-request-comments.yml]:
```suggestion
    name: Manage Trac ticket reminders for pull requests
```

File: .github/workflows/pull-request-comments.yml [hunk 2]
```diff
@@ -164,16 +164,16 @@ jobs:
 
             github.rest.issues.createComment( commentInfo );
 
-  # Leaves a comment on a pull request when no Trac ticket is included in the pull request description.
+  # Leaves a comment on a pull request when no Trac ticket is included in the pull request description and delete the comment if the Trac ticket has been updated.
```
Comment (desrosj) [inline .github/workflows/pull-request-comments.yml]:
```suggestion
  # Manages comments reminding contributors to include a Trac ticket link when opening a pull request.
  # 
  # This job will:
  # - Leave a comment on a pull request when no Trac ticket is included in the pull request description.
  # - Delete the comment once a Trac ticket has been added.
```

**Review by desrosj — CHANGES_REQUESTED** [review]
Thanks @Debarghya-Banerjee! I left a few small suggestions!

I've also been thinking through whether adding any event triggers would be appropriate. `pull_request_target` is required to ensure the workflow has the necessary capabilities to [manage comments and has the following `types`  by default](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target): `opened`, `synchronize`, and `reopened`.

We could probably add `edited` to this list. But all three would need to be explicitly listed.

File: .github/workflows/pull-request-comments.yml (hunk 1 above)
Comment (Debarghya-Banerjee) [reply .github/workflows/pull-request-comments.yml]: Fixed

Comment (Debarghya-Banerjee) [reply .github/workflows/pull-request-comments.yml]: Fixed

File: .github/workflows/pull-request-comments.yml (hunk 2 above)
Comment (Debarghya-Banerjee) [reply .github/workflows/pull-request-comments.yml]: Fixed

Comment (Debarghya-Banerjee) [reply .github/workflows/pull-request-comments.yml]: Fixed

**Comment by Debarghya-Banerjee** [comment]
Hi @desrosj, I have addressed the feedback and made the changes. I have also added the `synchronize`, `reopened`, & `edited` in the `pull_request_track`

I am re-assigning you for review. Thanks.

**Review by desrosj — APPROVED** [review]
Thanks @Debarghya-Banerjee! Let's give it a try!

**Comment by desrosj** [comment]
Merged in https://core.trac.wordpress.org/changeset/59169.

**Pull request closed without merging**